File: week1/Can-Yavuz(a.k.a=TuvaletPompasi)/twoSum.ts
function twoSum(nums: number[], target: number): number[] {
    const map: Record<number, number> = {};
 for (let i = 0; i < nums.length; i++) {
     const complement: number = target - nums[i];
     if (map[complement] !== undefined) {
         return [map[complement], i];
     }
     map[nums[i]] = i;
 }
 return [];
};

File: week1/8bitbully/twoSum.cpp
// 1. Two Sum
#include <unordered_map>
#include <vector>
using namespace std;

vector<int> twoSum(vector<int> &nums, int target) {
  unordered_map<int, int> hash_map;

  for (int i = 0; i < (int)nums.size(); i++) {

    int rem = target - nums[i];

    auto it = hash_map.find(rem);

    if (it != hash_map.end()) {
      return {it->second, i};
    }

    hash_map[nums[i]] = i;
  }

  return {};
}
